File: data-structures/node/ts/Node.ts
class Node {
  private value: number;
  private nextNode: Node | null;

  constructor(value: number, nextNode: Node | null = null) {
    this.value = value;
    this.nextNode = nextNode;
  }

  getValue(): number {
    return this.value;
  }

  getNextNode(): Node | null {
    return this.nextNode;
  }

  setNextNode(nextNode: Node | null): void {
    this.nextNode = nextNode;
  }
}

export { Node };

File: data-structures/node/Node.cpp
class Node {
private:
    int value;     // Node's value
    Node* nextNode; // Pointer to the next node

public:
    // Constructor
    Node(int val, Node* next = nullptr) {
        value = val;
        nextNode = next;
    }

    // Getter for value
    int getValue() const {
        return value;
    }

    // Getter for next node
    Node* getNextNode() const {
        return nextNode;
    }

    // Setter for next node
    void setNextNode(Node* next) {
        nextNode = next;
    }
};